### PR TITLE
Remove lodash

### PR DESCRIPTION
**Pull request by nirhart**

### CODE REVIEW

File: src/AutoCompleteInput/AutoCompleteInput.js [hunk 1]
```diff
@@ -42,7 +43,7 @@ class AutoCompleteInput extends React.Component {
 
           {this.props.header}
 
-          {_.map(this.props.suggestions, (suggestion, index) => {
+          {map(this.props.suggestions, (suggestion, index) => {
```
Comment (shahata) [inline src/AutoCompleteInput/AutoCompleteInput.js]: why not `this.props.sugestions.map((suggestion, index) => {`?

File: src/Slider/Slider.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ export default class Slider extends Component {
   }
 
   getMarks() {
-    return _.reduce(this.getRange(), (acc, cur) => {
+    return reduce(this.getRange(), (acc, cur) => {
```
Comment (shahata) [inline src/Slider/Slider.js]: why not `this.getRange().reduce((acc, cur) => {`?

File: src/Select/Select.js [hunk 3]
```diff
@@ -132,7 +134,7 @@ class Select extends React.Component {
           </svg>
         </div>
         <div data-testkey={(this.props.nameAttr) ? this.props.nameAttr + 'Select' : 'wixMenuSelect'} className={`${styles.options} ${this.state.open ? styles.shown : ''} ${this.props.dropDirectionUp ? styles.up : styles.down}`} ref="options">
-          {_.map(options, (option, idx) => (
+          {map(options, (option, idx) => (
```
Comment (shahata) [inline src/Select/Select.js]: options.map...

**Comment by 3LOK** [comment]
@nirhart, @shahata - I have only one major comments from all my comments (under the autocomplete part). Shahar - your comments are major or 'nice to have'?

**Comment by nirhart** [comment]
@3LOK I'm adding @shahata suggestions,
by major are you referring this - ' // TODO this method never gets called, this.props.suggestions[selectedSuggestion] is an object {text: 'fsdfds, ...} ' ?
IMO this is the only major issue

**Comment by 3LOK** [comment]
@nirhart - no, I meant:

> this isn't good. it means onMouseDown will re-render on each render.'

having something like:
```javascript
<div onClick={() => this.onClick()}></div>
```
means the div will __always__ get re-renderer as its onClick prop always changes. I'm not sure the previous solution did not have the same issue, but it's worth fixing if touching.

**Pull request merged**